### PR TITLE
Move LinearAlgebra.jl code into a package extension

### PR DESCRIPTION
**Pull request by MilesCranmer**
1. Moves LinearAlgebra.jl code into a package extension to reduce load time for packages which don't have it as a dependency.
2. Updates all three extensions to use Requires.jl so they get the same behavior on older Julia versions.
3. Bumps version to 1.5.5

### CODE REVIEW

File: ext/ConstructionBaseLinearAlgebraExt.jl [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+if isdefined(Base, :get_extension)
+    using ConstructionBase
+    using LinearAlgebra: LinearAlgebra, SubArray, LinRange, Cholesky, LowerTriangular, UpperTriangular, Tridiagonal
+else
+    using ..ConstructionBase
+    using ..LinearAlgebra: LinearAlgebra, SubArray, LinRange, Cholesky, LowerTriangular, UpperTriangular, Tridiagonal
+end
+
+### SubArray
```
Comment (rafaqz) [inline ext/ConstructionBaseLinearAlgebraExt.jl:11]: SubArray isnt in LinearAlgebra, neither is ReshapedArray, PermutedDimsArray, or LinRange...

**Review by rafaqz — CHANGES_REQUESTED** [review]
Thanks, good to make LinearAlgebra optional.

But only Cholesky and Tridiagonal are in LinearAlgebra, so can you move the other constructors back ?

File: ext/ConstructionBaseLinearAlgebraExt.jl (hunk 1 above)
Comment (MilesCranmer) [reply ext/ConstructionBaseLinearAlgebraExt.jl:11]: Oops, sorry.

Please feel free to edit directly to this PR. I almost posted an issue instead but I thought I might take a quick attempt at kicking off this PR.

Comment (rafaqz) [reply ext/ConstructionBaseLinearAlgebraExt.jl:11]: This is fine, it just needs those bits back where they were. But I will leave that to you ;)

**Comment by rafaqz** [comment]
No worries. To completely unlick the cookie: anyone reading this who needs it feel free to finish the PR!

**Pull request closed without merging**